src/components/Posts/Posts.tsx

```diff
@@ -111,6 +111,20 @@ export const Posts = ({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [currentPage, pages, sorting, filter]);
 
+  useEffect(() => {
+    const newFilter = {
+      category: params.get("category")?.split(",") || [],
+      initDate: params.get("initDate")
+        ? new Date(params.get("initDate")!)
+        : undefined,
+      finalDate: params.get("finalDate")
+        ? new Date(params.get("finalDate")!)
+        : undefined,
+    };
+
+    setFilter(newFilter);
+  }, [params]);
```

Comment:
Instead of using an useEffect here, what if we changed the filter from useState to useMemo? So it is updated every time the params changes?